analytics-api/src/analytics_api/models/user_response_detail.py

```diff
@@ -16,7 +16,7 @@ class UserResponseDetail(BaseModel):  # pylint: disable=too-few-public-methods
     __tablename__ = 'user_response_detail'
 
     id = db.Column(db.Integer, primary_key=True, nullable=False)
-    survey_id = db.Column(db.Integer, ForeignKey('survey.id', ondelete='CASCADE'), primary_key=True, nullable=False)
```

Comment:
I think you are missing the migration to remove the primary key?

Reply:
somehow its not there in the migration file and was only in the model.